maven-plugin/src/main/java/io/confluent/kafka/schemaregistry/maven/DownloadSchemaRegistryMojo.java

```diff
@@ -51,33 +49,48 @@
   @Parameter(required = true)
   List<String> subjectPatterns = new ArrayList<>();
 
+  @Parameter(required = false)
+  List<String> versions = new ArrayList<>();
+
   @Parameter(required = true)
   File outputDirectory;
 
-  Map<String, ParsedSchema> downloadSchemas(Collection<String> subjects)
+  Map<String, ParsedSchema> downloadSchemas(List<String> subjects, List<String> versionsToDownload)
       throws MojoExecutionException {
     Map<String, ParsedSchema> results = new LinkedHashMap<>();
 
-    for (String subject : subjects) {
+    for (int i = 0; i < subjects.size(); i++) {
       SchemaMetadata schemaMetadata;
       try {
-        getLog().info(String.format("Downloading latest metadata for %s.", subject));
-        schemaMetadata = this.client().getLatestSchemaMetadata(subject);
+        getLog().info(String.format("Downloading metadata "
+            + "for %s.for version %s", subjects.get(i), versionsToDownload.get(i)));
```

Comment:
This will break if `versionsToDownload` is empty

Reply:
versionsToDownload is a local array which will be populated with values "latest" if the input parameter "versions" is empty.